src/rdkafka_request.c

```diff
@@ -134,6 +134,7 @@ int rd_kafka_err_action(rd_kafka_broker_t *rkb,
                 break;
 
         case RD_KAFKA_RESP_ERR__TRANSPORT:
+        case RD_KAFKA_RESP_ERR__SSL:
```

Comment:
Not all the errors, the errors that happen during the handshake are treated differently in `rd_kafka_transport_ssl_handshake`. These are errors that happen when the connection is established in `rd_kafka_transport_ssl_io_update`.
They're handled by disconnecting from the broker and I think they have to be handled the same way an a transport error, by treating it as retriable instead of permanent.
In Java seems similar, the SSLException is treated differently in the handshake case but it's caught as an IOException if it happens when reading (`SslTransportLayer.State.READY`).